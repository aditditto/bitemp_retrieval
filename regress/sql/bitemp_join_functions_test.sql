DROP EXTENSION bitemp_retrieval;
CREATE EXTENSION bitemp_retrieval;

-- 10-----20
-- 10-----20
SELECT bitemp_timeperiod_joinable(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-20'::timestamptz)
);

SELECT bitemp_timeperiod_join(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-20'::timestamptz)
);

-- 10-----20
-- 10---15
SELECT bitemp_timeperiod_joinable(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-15'::timestamptz)
);

SELECT bitemp_timeperiod_join(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-15'::timestamptz)
);

-- 10-----20
--    15--20
SELECT bitemp_timeperiod_joinable(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-15'::timestamptz, '2020-10-20'::timestamptz)
);

SELECT bitemp_timeperiod_join(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-15'::timestamptz, '2020-10-20'::timestamptz)
);

-- 10-----20
--  11---19
SELECT bitemp_timeperiod_joinable(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-11'::timestamptz, '2020-10-19'::timestamptz)
);

SELECT bitemp_timeperiod_join(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-11'::timestamptz, '2020-10-19'::timestamptz)
);

-- 10---15
-- 10-----20
SELECT bitemp_timeperiod_joinable(
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-15'::timestamptz),
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz)
);

SELECT bitemp_timeperiod_join(
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-15'::timestamptz),
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz)
);

--    15--20
-- 10-----20
SELECT bitemp_timeperiod_joinable(
    temporal_relationships.timeperiod('2020-10-15'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz)
);

SELECT bitemp_timeperiod_join(
    temporal_relationships.timeperiod('2020-10-15'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz)
);

--  11---19
-- 10-----20
SELECT bitemp_timeperiod_joinable(
    temporal_relationships.timeperiod('2020-10-11'::timestamptz, '2020-10-19'::timestamptz),
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz)
);

SELECT bitemp_timeperiod_join(
    temporal_relationships.timeperiod('2020-10-11'::timestamptz, '2020-10-19'::timestamptz),
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz)
);

-- 10-----20
--    15-----25
SELECT bitemp_timeperiod_joinable(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-15'::timestamptz, '2020-10-25'::timestamptz)
);

SELECT bitemp_timeperiod_join(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-15'::timestamptz, '2020-10-25'::timestamptz)
);

--    15-----25
-- 10-----20
SELECT bitemp_timeperiod_joinable(
    temporal_relationships.timeperiod('2020-10-15'::timestamptz, '2020-10-25'::timestamptz),
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz)
);

SELECT bitemp_timeperiod_join(
    temporal_relationships.timeperiod('2020-10-15'::timestamptz, '2020-10-25'::timestamptz),
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz)
);

-- Not joinable
SELECT bitemp_timeperiod_joinable(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-11-10'::timestamptz, '2020-11-20'::timestamptz)
);

SELECT bitemp_timeperiod_joinable(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-20'::timestamptz, '2020-11-20'::timestamptz)
);