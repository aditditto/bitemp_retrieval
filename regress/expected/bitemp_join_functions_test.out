DROP EXTENSION bitemp_retrieval;
CREATE EXTENSION bitemp_retrieval;
-- 10-----20
-- 10-----20
SELECT bitemp_joinable(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-20'::timestamptz)
);
 bitemp_joinable 
-----------------
 t
(1 row)

SELECT bitemp_join(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-20'::timestamptz)
);
                           bitemp_join                           
-----------------------------------------------------------------
 ["Sat Oct 10 00:00:00 2020 PDT","Tue Oct 20 00:00:00 2020 PDT")
(1 row)

-- 10-----20
-- 10---15
SELECT bitemp_joinable(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-15'::timestamptz)
);
 bitemp_joinable 
-----------------
 t
(1 row)

SELECT bitemp_join(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-15'::timestamptz)
);
                           bitemp_join                           
-----------------------------------------------------------------
 ["Sat Oct 10 00:00:00 2020 PDT","Thu Oct 15 00:00:00 2020 PDT")
(1 row)

-- 10-----20
--    15--20
SELECT bitemp_joinable(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-15'::timestamptz, '2020-10-20'::timestamptz)
);
 bitemp_joinable 
-----------------
 t
(1 row)

SELECT bitemp_join(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-15'::timestamptz, '2020-10-20'::timestamptz)
);
                           bitemp_join                           
-----------------------------------------------------------------
 ["Thu Oct 15 00:00:00 2020 PDT","Tue Oct 20 00:00:00 2020 PDT")
(1 row)

-- 10-----20
--  11---19
SELECT bitemp_joinable(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-11'::timestamptz, '2020-10-19'::timestamptz)
);
 bitemp_joinable 
-----------------
 t
(1 row)

SELECT bitemp_join(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-11'::timestamptz, '2020-10-19'::timestamptz)
);
                           bitemp_join                           
-----------------------------------------------------------------
 ["Sun Oct 11 00:00:00 2020 PDT","Mon Oct 19 00:00:00 2020 PDT")
(1 row)

-- 10---15
-- 10-----20
SELECT bitemp_joinable(
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-15'::timestamptz),
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz)
);
 bitemp_joinable 
-----------------
 t
(1 row)

SELECT bitemp_join(
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-15'::timestamptz),
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz)
);
                           bitemp_join                           
-----------------------------------------------------------------
 ["Sat Oct 10 00:00:00 2020 PDT","Thu Oct 15 00:00:00 2020 PDT")
(1 row)

--    15--20
-- 10-----20
SELECT bitemp_joinable(
    temporal_relationships.timeperiod('2020-10-15'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz)
);
 bitemp_joinable 
-----------------
 t
(1 row)

SELECT bitemp_join(
    temporal_relationships.timeperiod('2020-10-15'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz)
);
                           bitemp_join                           
-----------------------------------------------------------------
 ["Thu Oct 15 00:00:00 2020 PDT","Tue Oct 20 00:00:00 2020 PDT")
(1 row)

--  11---19
-- 10-----20
SELECT bitemp_joinable(
    temporal_relationships.timeperiod('2020-10-11'::timestamptz, '2020-10-19'::timestamptz),
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz)
);
 bitemp_joinable 
-----------------
 t
(1 row)

SELECT bitemp_join(
    temporal_relationships.timeperiod('2020-10-11'::timestamptz, '2020-10-19'::timestamptz),
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz)
);
                           bitemp_join                           
-----------------------------------------------------------------
 ["Sun Oct 11 00:00:00 2020 PDT","Mon Oct 19 00:00:00 2020 PDT")
(1 row)

-- 10-----20
--    15-----25
SELECT bitemp_joinable(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-15'::timestamptz, '2020-10-25'::timestamptz)
);
 bitemp_joinable 
-----------------
 t
(1 row)

SELECT bitemp_join(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-15'::timestamptz, '2020-10-25'::timestamptz)
);
                           bitemp_join                           
-----------------------------------------------------------------
 ["Thu Oct 15 00:00:00 2020 PDT","Tue Oct 20 00:00:00 2020 PDT")
(1 row)

--    15-----25
-- 10-----20
SELECT bitemp_joinable(
    temporal_relationships.timeperiod('2020-10-15'::timestamptz, '2020-10-25'::timestamptz),
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz)
);
 bitemp_joinable 
-----------------
 t
(1 row)

SELECT bitemp_join(
    temporal_relationships.timeperiod('2020-10-15'::timestamptz, '2020-10-25'::timestamptz),
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz)
);
                           bitemp_join                           
-----------------------------------------------------------------
 ["Thu Oct 15 00:00:00 2020 PDT","Tue Oct 20 00:00:00 2020 PDT")
(1 row)

-- Not joinable
SELECT bitemp_joinable(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-11-10'::timestamptz, '2020-11-20'::timestamptz)
);
 bitemp_joinable 
-----------------
 f
(1 row)

SELECT bitemp_joinable(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-20'::timestamptz, '2020-11-20'::timestamptz)
);
 bitemp_joinable 
-----------------
 f
(1 row)

